daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -2360,6 +2354,45 @@ sealed abstract class DocumentPart(part: Node, parent: Document) {
 
 }
 
+object CanonData {
+  private lazy val doubleForwardPattern = "//.*".r
+  private lazy val openClosePattern = "(?s)/[*].*?[*]/".r
+  private lazy val noWarnCharsSet = "|()[].Xx \n\r"
+
+  /*
+  * Allow "//" and "/* */" to act as comments.
+  * Any valid XML characters not explicitly allowed are also considered comments and are removed.
+  */
+  def canonicalizeData(validCharactersSet: String, userData: String): String = {
+    var doWarning: Boolean = false
+    Assert.invariant(!userData.contains('\r'))      // \r should not exist in userData
```

Comment:
Won't this cause an invariant if the data contains `&#13x;`?

Reply:
Yeah, looks like those tests are failing because of this invariant. This line should be removed.

Reply:
Confirming you can put a CR in XML documents using &#13;. It is CR and CRLF as whitespace characters that get normalized into LF. 

So I believe when an Infoset is created from XML text (as by a TDML test that uses an XML infoset) it can contain CR via `&#13;` The Daffodil InfosetInputter does NOT convert these to LF. 

Interestingly, the daffodil XML infoset outputters used by parsing, those create XML and always they convert CRLF and CR to LF. So you cannot get a CR in an XML infoset from the daffodil parser.